### ISSUE TITLE
Comparing string with Request.Headers.ContentType when ContentType is null throws error

### ISSUE THREAD

**Issue by jeff-pang** [NancyFx/Nancy#2769]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of Nancy
- [x I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [x] I have searched [open](https://github.com/NancyFx/Nancy/issues) and [closed](https://github.com/NancyFx/Nancy/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
When applying equality operators with of Request.Headers.ContentType with a string when ContentType is null throws NullReferenceException

```C#
if (Request.Headers.ContentType == "application/xml")
```
### Steps to reproduce
Submit a HTTP GET with no content-type set which encounter the code above will result in NullReferenceException

### System Configuration
Net Core 1.1

- Nancy version:
Clinteastwood 
- Nancy host
  - [ ] ASP.NET
  - [x] OWIN
  - [ ] Self-Hosted
  - [ ] Other:
- Other Nancy packages and versions:
- Environment (Operating system, version and so on):
- .NET Framework version:
- Additional information:

**Comment by jeff-pang** [comment]
I think the expected behaviour should be that the equality fails rather than throw an exception. If in agreement I can take a stab at create a PR for this. Let me know

**Comment by thecodejunkie** [comment]
🤔 Yeah I think I agree

It is happening because if it does not find a header, it will fall back to `null` as seen in `RequestHeaders.cs`

```c#
public MediaRange ContentType
{
    get { return this.GetValue("Content-Type", x => new MediaRange(x.First()), null); }
    set { this.SetHeaderValues("Content-Type", value, x => new[] { x.ToString() }); }
}
```

Maybe it (and others in the file) should default to `string.Empty` instead of `null`. We do it for others like the `Host` header

```c#
public string Host
{
    get { return this.GetValue("Host", x => x.First(), string.Empty); }
    set { this.SetHeaderValues("Host", value, x => new[] { x }); }
}
```

Not sure why we have the inconsistency. 

Thoughs @NancyFx/most-valued-minions ? If we're OK with this change (which we should mark as `Breaking Change`) then @jeff-pang I'd be happy for us to take a pull-request

**Comment by khellang** [comment]
I think `string.Empty` makes sense here :+1:

**Comment by jeff-pang** [comment]
ok I'll create  PR for this

**Comment by jchannon** [comment]
agree & nice one Jeff

**Comment by youngbob** [comment]
The null reference exception is caused by the implicit cast to string when trying to compare a MediaRange to string. I fixed the implicit operator in #2889. This is no longer a breaking change since ContentType still returns null.
